app/src/main/java/com/example/drawingactivity/ColorUtils.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.drawingactivity
+
+import android.graphics.Color
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+object ColorUtils {
+
+    private val HEXADECIMAL_PATTERN: Pattern = Pattern.compile("\\p{XDigit}+")
+
+    /**
+     * A data class that contains color values.
+     * Contains: Hue, Saturation, Value, Red, Green, Blue, and Hex Value.
+     */
+    data class ColorState(
+        var hue: Int,
+        var saturation: Float,
+        var value: Float,
+        var red: Int,
+        var green: Int,
+        var blue: Int,
+        var hexCode: String
```

Comment:
you don't need the hexCode either if your getting the hue saturation and value from somewhere else, the hex can be generated similar to the rgbs from the hsv.

Reply:
Agreed, this is used to pass that information in a single object. I will reconsider if I hit some issues using it that way